custom_model_runner/datarobot_drum/drum/enum.py

```diff
@@ -311,3 +311,9 @@ class PayloadFormat:
     CSV = "csv"
     ARROW = "arrow"
     MTX = "mtx"
+
+
+class ExitCodes(Enum):
+    # This is the DRUM specific exit code. Please avoid using reserved/common exit codes. e.g.,
+    # 1, 2, 126, 127, 128, 128+n, 130, 225*
+    SCHEMA_VALIDATION_ERROR = 3  # used when the program exits due to custom task validation fails.
```

Comment:
Are we sure this will get thrown as a user code error in DataRobot?

Reply:
User exit code is used to indicate the program exit due to a specific root cause (e.g., schema validation error). Because we want to know why the ebj exist and raise the different exceptions accordingly. Do you have the suggestion on passing such info other than the standard exit code (0: success, 1: failure)  ?

Reply:
I'm okay with exit code 3, my concern is in DR. If a task fails with this exit code, will we mark it as an internal error or a user-code error. Can you run this locally and produce the schema validation error? And see if the custom task estimator throws `CustomModelFitInternalError` or `CustomModelUserCodeError`

Reply:
It raises `CustomTaskSchemaValueError` when the exit code is 3 in DR. It is in my other DR pr (https://github.com/datarobot/DataRobot/pull/90108/files#diff-1ed9168ba917a426fc4b3dbc27ea3311a12e5378753462163d2f8202fe1c3f9dR379-R392) 

This is an new exception designed for custom model schema validation error. It is similar with `CustomModelUserCodeOOMError` design.

Reply:
what will it raise now (before your PR gets merged) ?

Reply:
With the existing implementation of DR, 
- Raise CustomModelUserCodeOOMError if status == ExecutionStatus.oomerror.value
- Raise CustomModelUserCodeError if status != ExecutionStatus.completed.value